packages/patrol/android/src/main/kotlin/pl/leancode/patrol/Automator.kt

```diff
@@ -204,24 +204,36 @@ class Automator private constructor() {
 
         val uiSelector = UiSelector().className(EditText::class.java).instance(index)
         val uiObject = uiDevice.findObject(uiSelector)
-        uiObject.click()
+
+        if (showKeyboard) {
+            uiObject.click()
+        }
+
         uiObject.text = text
 
-        pressBack() // Hide keyboard.
+        if (showKeyboard) {
+            pressBack() // Hide keyboard.
+        }
     }
 
-    fun enterText(text: String, uiSelector: UiSelector, bySelector: BySelector, index: Int) {
+    fun enterText(text: String, uiSelector: UiSelector, bySelector: BySelector, index: Int, showKeyboard: Boolean) {
         Logger.d("enterText($text): $uiSelector, $bySelector")
 
         if (waitForView(bySelector, index) == null) {
             throw UiObjectNotFoundException("$uiSelector")
         }
 
         val uiObject = uiDevice.findObject(uiSelector).getFromParent(UiSelector().className(EditText::class.java))
-        uiObject.click()
+
+        if (showKeyboard) {
+            uiObject.click()
+        }
+
         uiObject.text = text
 
-        pressBack() // Hide keyboard.
+        if (showKeyboard) {
+            pressBack() // Hide keyboard.
```

Comment:
What if the app has a behavior, that after writing x characters app closes the keyboard. Won't this pressing back return the user to the previous screen? Shall we firstly check whether the keyboard is shown first?

Reply:
This sounds possible, but so far, nobody has reported problems with that. If they do, then we'll look into it.

Also, this is a "native" method - it's an escape hatch for when you cannot use Flutter's default `enterText()`.